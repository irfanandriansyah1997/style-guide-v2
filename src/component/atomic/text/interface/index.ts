import { Property } from 'csstype';
import { FC, HTMLAttributes } from 'react';

/**
 * Basic Component Props
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.05.12
 */
type IBasicComponentProps =
  | ITextParagraphType
  | ITextSpanType
  | ITextHeadingType;

/**
 * Basic Text Props
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.05.12
 */
export interface IBasicTextProps {
  color: string;
  fontFamily: string;
  fontSize: number;
  fontWeight: number;
  lineHeight: number;
  textAlign: Property.TextAlign;
  textDecoration: Property.TextDecoration;
}

/**
 * Text Component Export Default
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.05.12
 */
export type ITextExportDefault = FC<ITextProps> & {
  H1: FC<ITextWithTagsProps<ITextHeadingType>>;
  H2: FC<ITextWithTagsProps<ITextHeadingType>>;
  H3: FC<ITextWithTagsProps<ITextHeadingType>>;
  H4: FC<ITextWithTagsProps<ITextHeadingType>>;
  H5: FC<ITextWithTagsProps<ITextHeadingType>>;
  H6: FC<ITextWithTagsProps<ITextHeadingType>>;
  Paragraph: FC<ITextWithTagsProps<ITextParagraphType>>;
  Span: FC<ITextWithTagsProps<ITextSpanType>>;
};

export type ITextHeadingType = Omit<
  HTMLAttributes<HTMLHeadingElement>,
  'style'
>;

export type ITextParagraphType = Omit<
  HTMLAttributes<HTMLParagraphElement>,
  'style'
>;

/**
 * Text Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.05.12
 */
export type ITextProps = Partial<IBasicTextProps> &
  IBasicComponentProps & {
    tag: ITextType;
  };

export type ITextSpanType = Omit<HTMLAttributes<HTMLSpanElement>, 'style'>;

export type ITextType = 'p' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'span';

/**
 * Text With Tag Props Interface
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.05.12
 */
export type ITextWithTagsProps<Tags> = Partial<IBasicTextProps> & Tags;
