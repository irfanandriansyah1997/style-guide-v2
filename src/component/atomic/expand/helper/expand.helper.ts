import { objToString, verifiedIsNotEmpty } from '@99/helper';
import { Children, cloneElement, ReactElement, ReactNode } from 'react';

import {
  IExpandClassnameList as ClassnameList,
  IExpandClassnameList,
  IExpandContentProps,
  IExpandProps
} from '@/atomic/expand/interface';
import {
  ExpandContent,
  ExpandSelector
} from '@/atomic/expand/section/content.component';
import styles from '@/atomic/expand/style/style.module.scss';
import { checkClassnameAvailable } from '@/helper/component.helper';
import { PartialNull } from '@/interface/general';

interface IExpandChildren {
  content: ReactElement[];
  selector: ReactElement[];
}

/**
 * Generate Classname Container Element
 * @param {Partial<IExpandProps>} props - expand component props
 * @returns {string}
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.06.06
 */
export const generateClassnameContainer = ({
  className,
  selectorPosition
}: Partial<IExpandProps>): string => {
  const response = {
    flex: true,
    'flex-column': selectorPosition === `top`,
    'flex-column-reverse': selectorPosition === `bottom`,
    [styles[`a-expand`]]: true,
    [`${className?.wrapper}`]: checkClassnameAvailable<ClassnameList>(
      className,
      `wrapper`
    )
  };

  return objToString(response);
};

/**
 * Transform Children To Expand Content
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.06.06
 */
export const generateContent = (
  children: ReactNode,
  classNameProps: PartialNull<IExpandClassnameList>
): ReactNode => {
  const response: ReactElement[] = [];
  const { content, selector } = Children.map(
    children as ReactElement,
    (child: ReactElement) => {
      const { type } = child as ReactElement;

      if ([ExpandContent, ExpandSelector].includes(type as any)) {
        return child;
      }

      return null;
    }
  )
    .filter(verifiedIsNotEmpty)
    .reduce(
      (prev: IExpandChildren, current): IExpandChildren => {
        const { type } = current as ReactElement;

        if ([ExpandContent].includes(type as any)) {
          return {
            ...prev,
            content: [...prev.content, current]
          };
        }

        return {
          ...prev,
          selector: [...prev.content, current]
        };
      },
      {
        content: [],
        selector: []
      }
    );

  if (selector.length > 0) response.push(selector[0]);
  if (content.length > 0) response.push(content[0]);

  return response.map((item, index) => {
    const { props } = item;
    const { className, ...res } = props as IExpandContentProps;

    return cloneElement(item, {
      ...res,
      className: objToString({
        [`${classNameProps?.toggle}`]:
          index === 0 &&
          checkClassnameAvailable<ClassnameList>(classNameProps, `toggle`),
        [`${classNameProps?.content}`]:
          index === 1 &&
          checkClassnameAvailable<ClassnameList>(classNameProps, `content`),
        [`${className}`]: verifiedIsNotEmpty(className)
      })
    });
  });
};
