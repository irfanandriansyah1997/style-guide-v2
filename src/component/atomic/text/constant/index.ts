import { IStorybookArgsItem } from '@/modules/storybook/interface';

export const CUSTOM_ARGS_TEXT_COMPONENT: IStorybookArgsItem[] = [
  {
    argsName: `children`,
    control: {
      type: `text`
    },
    defaultValue: ``,
    description: `Content on text component`,
    table: {
      category: `Content`,
      type: {
        summary: `ReactNode`
      }
    },
    type: {
      required: true
    }
  },
  {
    argsName: `color`,
    control: {
      type: `color`
    },
    description: `What text color to use`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `string`
      }
    }
  },
  {
    argsName: `fontFamily`,
    control: {
      type: `text`
    },
    defaultValue: `'Museo Sans', sans-serif`,
    description: `What typeface to use`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `string`
      }
    }
  },
  {
    argsName: `fontSize`,
    control: {
      max: 64,
      min: 12,
      type: `range`
    },
    defaultValue: `-`,
    description: `CSS property sets the size of the font`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `number`
      }
    }
  },
  {
    argsName: `fontWeight`,
    control: {
      options: [100, 200, 300, 400, 500, 600, 700],
      type: `select`
    },
    defaultValue: 400,
    description: `CSS property sets the weight (or boldness) of the font`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `number`
      }
    }
  },
  {
    argsName: `lineHeight`,
    control: {
      max: 64,
      min: 12,
      type: `range`
    },
    defaultValue: `-`,
    description: `CSS property sets the height of a line box`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `number`
      }
    }
  },
  {
    argsName: `textAlign`,
    control: {
      options: [
        `-moz-initial`,
        `inherit`,
        `initial`,
        `revert`,
        `unset`,
        `center`,
        `end`,
        `justify`,
        `left`,
        `match-parent`,
        `right`,
        `start`
      ],
      type: `select`
    },
    description: `CSS property sets the horizontal alignment of a block element`,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `Property.TextAlign`
      }
    }
  },
  {
    argsName: `textDecoration`,
    control: {
      options: [
        `auto`,
        `blink`,
        `dashed`,
        `dotted`,
        `double`,
        `from-font`,
        `grammar-error`,
        `line-through`,
        `none`,
        `overline`,
        `solid`,
        `spelling-error`,
        `underline`,
        `wavy`
      ],
      type: `select`
    },
    description: `CSS property sets the color of decorations added to text by \`text-decoration-line\``,
    table: {
      category: `Styling`,
      defaultValue: {
        summary: `-`
      },
      type: {
        summary: `Property.TextDecoration`
      }
    }
  }
];
