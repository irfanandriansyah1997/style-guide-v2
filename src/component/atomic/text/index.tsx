import BasicTextComponent from './section/basic.component';
import {
  Heading1Component,
  Heading2Component,
  Heading3Component,
  Heading4Component,
  Heading5Component,
  Heading6Component
} from './section/heading.component';
import ParagraphComponent from './section/paragraph.component';
import SpanComponent from './section/span.component';
import { ITextExportDefault } from './interface';

/**
 * Text Component
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.05.12
 */
const TextComponent: ITextExportDefault = BasicTextComponent as ITextExportDefault;

TextComponent.H1 = Heading1Component;

TextComponent.H2 = Heading2Component;

TextComponent.H3 = Heading3Component;

TextComponent.H4 = Heading4Component;

TextComponent.H5 = Heading5Component;

TextComponent.H6 = Heading6Component;

TextComponent.Paragraph = ParagraphComponent;

TextComponent.Span = SpanComponent;

export default TextComponent;
