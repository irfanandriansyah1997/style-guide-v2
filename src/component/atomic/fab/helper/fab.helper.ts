import { objToString, verifiedIsNotEmpty } from '@99/helper';
import { CSSProperties } from 'react';
import { TransitionStatus } from 'react-transition-group';

import { IFabAnimation, IFabStyling } from '@/atomic/fab/interface';
import animationStyles from '@/atomic/fab/style/animation.module.scss';
import styles from '@/atomic/fab/style/style.module.scss';

/**
 * Generate FAB Animation while fade in & fade out
 * @param {IFabAnimation | undefined} animation - animation className
 * @param {TransitionStatus} state - transtion state
 * @return {string}
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.14
 */
export const generateFABAnimation = (
  animation: IFabAnimation | undefined,
  state: TransitionStatus
): string => {
  let animationClassname = ``;
  if (verifiedIsNotEmpty(animation)) {
    switch (state) {
      case `entered`:
        animationClassname = `${animationStyles[`${animation}-entered`]}`;
        break;

      case `entering`:
        animationClassname = `${animationStyles[`${animation}-entering`]}`;
        break;

      case `exited`:
        animationClassname = `${animationStyles[`${animation}-exited`]}`;
        break;

      case `exiting`:
        animationClassname = `${animationStyles[`${animation}-exiting`]}`;
        break;

      default:
        throw new Error(`Unknown state`);
    }
  }

  return animationClassname;
};

/**
 * Generate FAB Classname
 * @param {IFabStyling} styling - fab styling props
 * @param {string | undefined} className - custom classname
 * @returns {string}
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.14
 */
export const generateFABClassName = (
  { position, rtl, theme, type }: Partial<IFabStyling>,
  className: string | undefined
): string =>
  objToString({
    flex: true,
    'flex-align-center': true,
    'flex-justify-center': true,
    'flex-row': true,
    'flex-row-reverse': rtl,
    [`${position}`]: verifiedIsNotEmpty(position),
    [styles[`a-fab`]]: true,
    [styles[`a-fab--type-${type}`]]: true,
    [styles[`a-fab--theme-${theme}`]]: true,
    [`${className}`]: verifiedIsNotEmpty(className)
  });

/**
 * Generate FAB Style
 * @param {IFabStyling} styling - fab styling props
 * @returns {CSSProperties}
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.07.14
 */
export const generateFABStyle = ({
  bottom,
  left,
  position,
  right,
  top
}: Partial<IFabStyling>): CSSProperties => ({
  bottom,
  left,
  position,
  right,
  top
});
