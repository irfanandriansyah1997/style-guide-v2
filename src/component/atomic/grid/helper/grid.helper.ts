import { objToString, verifiedIsNotEmpty } from '@99/helper';
import { CSSProperties } from 'react';

import {
  IBasicSizeGrid,
  IGridRowStyling,
  IGridSize,
  IGridSpaceItem
} from '@/atomic/grid/interface';
import styles from '@/atomic/grid/style/style.module.scss';

/**
 * Generate Spacing Size
 * @param {IGridSpaceItem | number} param - spacing size props
 * @returns {IGridSpaceItem}
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.06.16
 */
const generateSpacingSize = (
  param?: Partial<IGridSpaceItem> | number
): IGridSpaceItem => {
  const defaultSize = 12;

  if (!verifiedIsNotEmpty(param)) {
    return {
      horizontal: defaultSize,
      vertical: defaultSize
    };
  }

  if (typeof param === `number`) {
    return {
      horizontal: param / 2,
      vertical: param / 2
    };
  }

  return {
    horizontal: ((param as IGridSpaceItem).horizontal || defaultSize) / 2,
    vertical: ((param as IGridSpaceItem).vertical || defaultSize) / 2
  };
};

/**
 * Generate Class Name Wrapper
 * @param {string | undefined} className - custom classname
 * @returns {string}
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.06.16
 */
export const generateClassNameItem = (
  size: IGridSize,
  { lg, md, sm, xl, xs }: Partial<IBasicSizeGrid>,
  className?: string
): string =>
  objToString({
    [`${className}`]: verifiedIsNotEmpty(className),
    [`${styles[`a-col`]}`]: size === `auto`,
    [`${styles[`a-col-${size}`]}`]: size !== `auto`,
    [`${styles[`a-col-xs-${xs}`]}`]: verifiedIsNotEmpty(xs) && xs !== `auto`,
    [`${styles[`a-col-xs`]}`]: verifiedIsNotEmpty(xs) && xs === `auto`,
    [`${styles[`a-col-sm`]}`]: verifiedIsNotEmpty(sm) && sm === `auto`,
    [`${styles[`a-col-sm-${sm}`]}`]: verifiedIsNotEmpty(sm) && sm !== `auto`,
    [`${styles[`a-col-md`]}`]: verifiedIsNotEmpty(md) && md === `auto`,
    [`${styles[`a-col-md-${md}`]}`]: verifiedIsNotEmpty(md) && md !== `auto`,
    [`${styles[`a-col-lg`]}`]: verifiedIsNotEmpty(lg) && lg === `auto`,
    [`${styles[`a-col-lg-${lg}`]}`]: verifiedIsNotEmpty(lg) && lg !== `auto`,
    [`${styles[`a-col-xl`]}`]: verifiedIsNotEmpty(xl) && xl === `auto`,
    [`${styles[`a-col-xl-${xl}`]}`]: verifiedIsNotEmpty(xl) && xl !== `auto`
  });

/**
 * Generate Class Name Wrapper
 * @param {string | undefined} className - custom classname
 * @returns {string}
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.06.16
 */
export const generateClassNameWrapper = (className?: string): string =>
  objToString({
    flex: true,
    'flex-wrap': true,
    [`${className}`]: verifiedIsNotEmpty(className)
  });

/**
 * Generate Style Wrapper
 * @param {IGridRowStyling} props - props styling
 * @returns {CSSProperties}
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.06.16
 */
export const generateStyleItem = (
  space: Partial<IGridSpaceItem> | number | undefined,
  order: number | undefined
): CSSProperties => {
  const { horizontal, vertical } = generateSpacingSize(space);

  return {
    order,
    paddingBottom: vertical,
    paddingLeft: horizontal,
    paddingRight: horizontal,
    paddingTop: vertical
  };
};

/**
 * Generate Style Wrapper
 * @param {IGridRowStyling} props - props styling
 * @returns {CSSProperties}
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.06.16
 */
export const generateStyleWrapper = ({
  alignItems,
  justifyContent,
  spaceEachItem
}: IGridRowStyling): CSSProperties => {
  const { horizontal, vertical } = generateSpacingSize(spaceEachItem);

  return {
    alignItems,
    justifyContent,
    marginBottom: vertical,
    marginLeft: -1 * horizontal,
    marginRight: -1 * horizontal,
    marginTop: -1 * vertical
  };
};
