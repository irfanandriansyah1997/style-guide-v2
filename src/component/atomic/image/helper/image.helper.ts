import { verifiedIsNotEmpty, verifiedKeyIsExist } from '@99/helper';
import { ReactElement, ReactNode } from 'react';

import { IImagePresetProps } from '@/atomic/image/interface';
import { ImagePreset } from '@/atomic/image/section/image-responsive.component';
import { transformChildrenToArray } from '@/helper/component.helper';
import { IImageResponsive } from '@/interface/general';

/**
 * Transform Children To Image Responsive Object
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.06.06
 */
export const transformChildrenToImageResponsive = (
  children: ReactNode
): Partial<IImageResponsive> => {
  const props = transformChildrenToArray<IImagePresetProps>(children)(
    (child) => {
      if (
        verifiedIsNotEmpty(child) &&
        verifiedKeyIsExist((child as unknown) as Record<string, any>, `type`)
      ) {
        const { props, type } = child as ReactElement;

        if ([ImagePreset].includes(type as any)) {
          return props;
        }
      }

      return undefined;
    }
  ).sort((a, b) => a.sizes - b.sizes);

  return {
    src: props.length > 0 ? props[0].src : undefined,
    srcSet: (props || []).map(({ sizes, src }) => `${src} ${sizes}w`).join(`,`)
  };
};
