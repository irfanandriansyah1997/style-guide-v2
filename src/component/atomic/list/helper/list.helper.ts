import {
  objToString,
  verifiedIsNotEmpty,
  verifiedKeyIsExist
} from '@99/helper';
import { CSSProperties, Key, ReactElement, ReactNode } from 'react';

import {
  IListContent,
  IListContentType,
  IListItemProps,
  IListStyle
} from '@/atomic/list/interface';
import ListItem from '@/atomic/list/section/list-item.component';
import { transformChildrenToArray } from '@/helper/component.helper';

/**
 * Generate Classname Container
 * @param {Partial<IListStyle> | undefined} style - style object props
 * @returns {string}
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.06.11
 */
export const generateClassnameContainer = (
  style: Partial<IListStyle> | undefined
): string => {
  let className: Record<string, boolean> = {
    flex: true,
    'flex-wrap': true,
    relative: true
  };
  if (style) {
    const { orientation } = style;

    switch (orientation) {
      case `horizontal`:
        className = {
          ...className,
          'flex-align-center': true,
          'flex-row': true
        };
        break;
      case `vertical`:
        className = {
          ...className,
          'flex-align-start': true,
          'flex-column': true
        };
        break;
    }
  }

  return objToString(className);
};

/**
 * Generate Style Item
 * @param {Partial<IListStyle> | undefined} style - style object props
 * @returns {CSSProperties}
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.06.11
 */
export const generateStyleItem = (
  style: Partial<IListStyle> | undefined
): CSSProperties => {
  if (style) {
    const { dividerColor: borderColor, orientation, outline, space } = style;
    let padding: string | undefined;
    let borderStyle: string | undefined;

    if (space) {
      switch (orientation) {
        case `horizontal`:
          padding = `0px ${(space as number) / 2}px`;
          break;
        case `vertical`:
          padding = `${(space as number) / 2}px 0px`;
          break;
      }
    }

    if (outline) {
      switch (outline) {
        case `solid`:
          borderStyle = `solid`;
          break;
        case `dash`:
          borderStyle = `dashed`;
          break;
      }
    }

    return {
      borderColor,
      borderStyle,
      padding
    };
  }

  return {};
};

/**
 * Transform Children To List Item
 * @param {ReactNode} children - children Props
 * @returns {IBreadcrumbItemProps}
 * @author Irfan Andriansyah <irfan@99.co>
 * @since 2021.06.11
 */
export const transformChildrenToListContent = (
  children: ReactNode
): IListContent[] =>
  transformChildrenToArray<{
    item: IListItemProps;
    key: Key | null;
  }>(children)((child) => {
    if (
      verifiedIsNotEmpty(child) &&
      verifiedKeyIsExist((child as unknown) as Record<string, any>, `type`)
    ) {
      const { key, props, type } = child as ReactElement;

      if ([ListItem].includes(type as any)) {
        return {
          item: props,
          key
        };
      }
    }

    return undefined;
  }).map(
    ({ item, key }, index): IListContent => ({
      key: key ? `${key}` : `${index}`,
      payload: item,
      type: IListContentType.item
    })
  );
