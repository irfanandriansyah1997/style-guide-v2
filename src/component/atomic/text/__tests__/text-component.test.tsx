import { render, screen } from '@testing-library/react';

import Text from '@/atomic/text';
import {
  STYLE_GUIDE_FONT_SIZE,
  STYLE_GUIDE_FONT_WEIGHT,
  STYLE_GUIDE_LINE_HEIGHT
} from '@/constant/typography';
import { doTestFixture } from '@/helper/unit-testing.helper';

import { FIXTURE_TEXT_COMPONENT } from './__mocks__/text.mock';

describe(`Test text component`, () => {
  it(`Render text component normally`, () => {
    render(
      <Text
        tag="p"
        lineHeight={STYLE_GUIDE_LINE_HEIGHT[`caption-1`]}
        fontSize={STYLE_GUIDE_FONT_SIZE[`caption-1`]}
        fontWeight={STYLE_GUIDE_FONT_WEIGHT.light}
      >
        Hello
      </Text>
    );
    const element = screen.getByText(`Hello`);

    expect(element.className).toBe(`a-text`);
    expect(element.style.fontSize).toBe(`12px`);
    expect(element.style.fontWeight).toBe(`300`);
    expect(element.style.lineHeight).toBe(`16px`);
    expect(element.tagName).toBe(`P`);
  });

  it(`Render text component without line height`, () => {
    const { container } = render(
      <Text
        tag="p"
        className="test"
        fontSize={STYLE_GUIDE_FONT_SIZE[`caption-1`]}
        fontWeight={STYLE_GUIDE_FONT_WEIGHT.light}
      >
        Hello
      </Text>
    );
    const element = screen.getByText(`Hello`);

    expect(element.className).toBe(`a-text test`);
    expect(element.style.fontSize).toBe(`12px`);
    expect(element.style.fontWeight).toBe(`300`);
    expect(element.style.lineHeight).toBe(``);
    expect(element.tagName).toBe(`P`);
    expect(container.outerHTML).toMatchSnapshot();
  });

  it(`Testing With Mock`, () => {
    doTestFixture(
      FIXTURE_TEXT_COMPONENT,
      (
        input,
        {
          color,
          fontFamily,
          fontSize,
          fontWeight,
          lineHeight,
          tag,
          textAlign,
          textDecoration
        }
      ) => {
        const { container } = render(<Text {...input}>Hello</Text>);
        const element = container.children;

        if (element.length > 0) {
          expect((element[0] as HTMLElement).style.color).toBe(color);
          expect((element[0] as HTMLElement).style.fontFamily).toBe(fontFamily);
          expect((element[0] as HTMLElement).style.fontSize).toBe(fontSize);
          expect((element[0] as HTMLElement).style.fontWeight).toBe(fontWeight);
          expect((element[0] as HTMLElement).style.lineHeight).toBe(lineHeight);
          expect((element[0] as HTMLElement).tagName).toBe(tag);
          expect((element[0] as HTMLElement).style.textAlign).toBe(textAlign);
          expect((element[0] as HTMLElement).style.textDecoration).toBe(
            textDecoration
          );
          expect(container.outerHTML).toMatchSnapshot();
        }
      }
    );
  });
});
