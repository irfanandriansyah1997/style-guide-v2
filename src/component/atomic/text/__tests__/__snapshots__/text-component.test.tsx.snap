// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test text component Render text component without line height 1`] = `"<div><p class=\\"a-text test\\" style=\\"font-size: 12px; font-weight: 300;\\">Hello</p></div>"`;

exports[`Test text component Testing With Mock 1`] = `"<div><p class=\\"a-text\\" style=\\"color: rgb(255, 232, 214); font-family: a; font-size: 12px; font-weight: 900; line-height: 16px; text-align: center; text-decoration: underline;\\">Hello</p></div>"`;

exports[`Test text component Testing With Mock 2`] = `"<div><span class=\\"a-text\\" style=\\"color: rgb(255, 247, 209); font-family: b; font-size: 12px; font-weight: 300; line-height: 16px;\\">Hello</span></div>"`;

exports[`Test text component Testing With Mock 3`] = `"<div><h1 class=\\"a-text\\" style=\\"color: rgb(255, 247, 209); font-family: b; line-height: 16px;\\">Hello</h1></div>"`;
